src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +163,58 @@ class bo : public xrt::bo
   }
 };
 
+class event_impl;
+class event 
+{
+public:
+
+  /**
+   * event() - Constructor from a device
+   *
+   * @param device
+   *  The device on which the profiling event should start
+   *
+   */
+  event(const xrt::device& device);
+
+  /**
+   * start_profiling() - Start AIE performance profiling
+   *
+   * @param option
+   *  Profiling option
+   * @param port1_name
+   *  Profiling port 1 name.
+   * @param port2_name
+   *  Profiling port 2 name.
+   * @param value
+   *  The number of bytes to trigger the profiling event.
+   *
+   * This function configures the performance counters in AI Engine by given
+   * port names and value. The port names and value will have different
+   * meanings on different options.
+   */
+  int 
+  start_profiling(int option, const std::string& port1_name, const std::string& port2_name, uint32_t value) const;
+
+  /**
+   * read_profiling() - Read the current performance counter value
+   *                    associated with the profiling handle
+   */
+  uint64_t
+  read_profiling() const;
+
+  /**
+   * stop_profiling() - Stop the current performance profiling
+   *                    associated with the profiling handle and
+   *                    release the corresponding hardware resources.
+   */
+  void
+  stop_profiling() const;
+
+private:
+    std::shared_ptr<event_impl> impl;
```

Comment:
If this is meant to be a public class, then maybe use `core/include/xrt/detail/pimpl.h` as base class; look at other examples.